util.go

```diff
@@ -2,24 +2,46 @@ package cuckoo
 
 import (
 	"encoding/binary"
-	"math/rand"
+	"unsafe"
 
-	metro "github.com/dgryski/go-metro"
+	"github.com/zeebo/wyhash"
+	"github.com/zeebo/xxh3"
+)
+
+var (
+	// plus 1 to prevent out of range.
+	altHash = [maxFingerprint + 1]uint{}
+	rng     wyhash.SRNG
 )
 
 // randi returns either i1 or i2 randomly.
 func randi(i1, i2 uint) uint {
-	if rand.Int31()%2 == 0 {
+	// it's faster than mod, but the result is the almost same.
+	if uint32(uint64(uint32(rng.Uint64()))*uint64(2)>>32) == 0 {
```

Comment:
This makes it much harder to read, is the optimization really worth it? Please do a separate PR/evaluation.

Reply:
After studying the code in rng.Uint64(), I see that there will be a speed up in particular on concurrent use. So moving to the new library looks good to me.

But I don't like the obfuscated code here. Please revert it to

rng.Uint64() % 2 == 0

In my opinion, readability is more important here than the small gain from converting and bit shifting.